extensions/mssql/src/objectExplorerNodeProvider/fileSources.ts

```diff
@@ -147,18 +150,43 @@ export class FileSourceFactory {
 }
 
 export class HdfsFileSource implements IFileSource {
+	private mounts: Map<string, Mount>;
 	constructor(private client: WebHDFS) {
 	}
 
-	public enumerateFiles(path: string): Promise<IFile[]> {
+	public async enumerateFiles(path: string, refresh?: boolean): Promise<IFile[]> {
+		if (!this.mounts || refresh) {
+			await this.loadMounts();
+		}
+		return this.readdir(path);
+	}
+
+	private loadMounts(): Promise<void> {
+		return new Promise((resolve, reject) => {
```

Comment:
With these changes, is our automation around promises being used correctly happy?

Not saying that I\'m spotting anything necessarily, just want to make sure that we're good there.

Reply:
What does the automation do again? This is the same basic pattern as previous - we use Promises, and this is mostly converting callbacks into those?

Reply:
Verified this shouldn't be impacted